pkg/kv/kvserver/replica_batch_updates.go

```diff
@@ -239,8 +239,11 @@ func tryBumpBatchTimestamp(
 	}
 	log.VEventf(ctx, 2, "bumping batch timestamp to: %s from read: %s, write: %s",
 		ts, ba.Txn.ReadTimestamp, ba.Txn.WriteTimestamp)
-	ba.Txn = ba.Txn.Clone()
-	ba.Txn.BumpReadTimestamp(ts)
-	ba.Timestamp = ba.Txn.ReadTimestamp // Refresh just updated ReadTimestamp
+	txn := ba.Txn.Clone()
+	txn.BumpReadTimestamp(ts)
+	readTs := ba.Txn.ReadTimestamp
+	ba = ba.ShallowCopy()
```

Comment:
Thanks for guidance, I am trying to understand this pattern - the data race happens a pointer level, we return a new shallow copy pointer on mutate to prevent race on the same pointer down the line. But this approach cannot prevent the race in the value in memory right?